docs/docs/examples-tutorials/recipes/running-tracetest-with-aws-terraform.md

```diff
@@ -0,0 +1,266 @@
+
+### 3. Jaeger
+
+Inside the `jaeger.tf` file you'll find the required services to run the all in one instance using AWS Fargate.
+
+### AWS Network
+
+To connect all of the services, the example generates a VPC network which includes private and public subnets. Internal service like the Postgres RDS instance are protected behind the VPC and only accessible through a node within the internal network.
+
+There are two [Application Load Balancers](https://aws.amazon.com/elasticloadbalancing/application-load-balancer/), a public balancer which provides access to the Tracetest task instance through port `11633` and to the Jaeger UI from `16686`.
+
+The second is an internal load balancer that allows the instrumented Node.js Lambda functions connect to the Jaeger OTLP endpoint using a dns instead of an IP.
+
+## Node.js Serverless API
+
+The Node.js Serverless API is a simple Lambda function, contained in the `src/hello.js` file.
+
+The OpenTelemetry tracing is contained in the `src/tracing.js` file.
+Traces will be sent to the Jaeger OpenTelemetry Collector Endpoint.
+
+Here's the content of the `tracing.js` file:
+
+```js
+const { Resource } = require("@opentelemetry/resources");
+const api = require("@opentelemetry/api");
+const { BatchSpanProcessor } = require("@opentelemetry/sdk-trace-base");
+const { OTLPTraceExporter } = require("@opentelemetry/exporter-trace-otlp-http");
+const { NodeTracerProvider } = require("@opentelemetry/sdk-trace-node");
+const { registerInstrumentations } = require("@opentelemetry/instrumentation");
+const { getNodeAutoInstrumentations } = require("@opentelemetry/auto-instrumentations-node");
+const { SemanticResourceAttributes } = require("@opentelemetry/semantic-conventions");
+
+const provider = new NodeTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: "tracetest",
+  }),
+});
+const spanProcessor = new BatchSpanProcessor(new OTLPTraceExporter());
+
+provider.addSpanProcessor(spanProcessor);
+provider.register();
+
+registerInstrumentations({
+  instrumentations: [
+    getNodeAutoInstrumentations({
+      "@opentelemetry/instrumentation-aws-lambda": {
+        disableAwsContextPropagation: true,
+        eventContextExtractor: (event) => {
+          event.headers.traceparent = event.headers.Traceparent;
+          const eventContext = api.propagation.extract(api.context.active(), event.headers);
+
+          return eventContext;
+        },
+      },
+    }),
+  ],
+});
+```
+
+The hostname for the Jaeger OTLP Endpoint will be:
+
+- HTTP: `http://<internal_load_balancer_dns_name>:4318`
+
+Enabling the tracer is done by adding an environment variable to the lambda function.
+
+```bash
+NODE_OPTIONS="--require tracing.js"
+```
+
+## Tracetest
+
+The `tracetest.tf` file contains the required services for the Tracetest server which include.
+
+- **Postgres RDS** - Postgres is a prerequisite for Tracetest to work. It stores trace data when running the trace-based tests.
```

Comment:
- **Postgres RDS** - Postgres is a prerequisite for Tracetest to work. It stores the trace-based tests you create, information about prior test runs, and other data that Tracetest needs.